docs/source/intro.rst

```diff
@@ -7,19 +7,34 @@ Taweret for Bayesian Model Mixing
 Bayesian Model Mixing
 ---------------------
 
-In computer simulation studies we often come across multiple theoretically \
-sound models that are equally good. These models have different assumptions built into them \
-and are only valid in certain regions of the input domain. It is most certainly possible that the true \
-model that can describe the phenomena is not any of the models that are being considered. Bayesian Model \
-Mixing is a data-driven technique that allows us to combine the models to get the most accurate predictions\
-and calibrations while taking into account the modeling uncertainties. Taweret is a python package \
-for Bayesian Model Mixing. 
-
-Typical workflow of Bayesian Model Mixing
-
-1. Collect all the models and experimental data 
-2. Decide on a mixing method
-3. Find the weights/mixing function to combine models
+In computer simulation studies, it is often the case where a set of theoretically \
+sound models are considered, each intended to describe the true underlying physical phenomena of interest \
+across a sub-region of the domain. These models could differ by their underlying assumptions \
+and may not be valid in certain sub-regions of the input domain. In practice, the true underlying \
+model is not contained in the set of candidate models. Thus, selecting a single model to to describe the true phenomena \
+across the entire input domain is inappropriate. As an alternative, one may elect to combine the information within \
+the model set in some systematic manner. A common approach involves combining the individual \
+mean predictions or predictive densities from the indivdual models using a linear combination or weight4ed average. \
+The weights in this linear combination may or may not depend on the inputs. When the models under consideration \
+exhibit varrying levels of predictive accuracy depending on the sub-region of the input domain, an input-dependent \
+weighting scheme is more appropriate. A memeber of the class of input-dependent weighting schemes is \
+Bayesian Model Mixing (BMM). BMM is a data-driven technique combines the predictions from a set of N candidate models in a \
+Bayesian manner using input-dependent weights. Mixing can be performed using one of two strategies described below: \
+(1) A two-step approach: Each model is fit using one source of data prior to mixing. \
+Conditional on the predictions from each model, the weight functions are learned using a second source of data. \
```

Comment:
Why do weight have to be learned from a second source? For example in the Coleman thesis BMM example he used the same data for model calibration and then the mixing of models.

Reply:
If you are doing a two step approach meaning (1) fit each model (2) given each fitted model then learn the weights... then you must use two sources of data for each step (or use some clever cross validation ideas). Otherwise you would be reusing data in both steps, which is typically a problem (causes overfitting). Can you forward me the Coleman thesis? I will take a look and get a better understanding about his method.

Reply:
https://dukespace.lib.duke.edu/dspace/handle/10161/18788

Reply:
I'm use the same experimental data that was used for calibration in the next step of learning the weights in my physics application. Experimental data is a luxury to us and I don't see a way around it for my use case. If there is no other strong theoretical reasoning behind it, I think we can ignore the risk of overfitting and make this definition general.

Reply:
My understanding of the Coleman work is he was doing a joint analysis, meaning the models were being fit while simultaneously learning the weight functions. If you use this joint approach, then you can use the same data set!

Reply:
I'll read this excerpt from the thesis and get back to you shortly

Reply:
This looks like he is defining a joint model, meaning the mixing function parameters (beta0 and beta1) and the calibration parameters (theta1 and theta2) are being learned at the same time. In such case, you only need one source of data. 
![coleman_model_pg70](https://user-images.githubusercontent.com/40618811/213798023-d4e57157-0bdc-4d23-b33b-00433430331c.PNG)

Reply:
Yes. He does not use calibrated models but he does the calibration and mixing together here. There is no example for the other case where mixing of calibrated models happen. 

I wonder what can we do when the data is limited and we do not have a two sources of data , one for calibration and then one for mixing of the calibrated models. I don't see any other option than using the same data for leaning the mixing weights. And by adding "weight functions are learned using a second source of data" I think we are being too restrictive.  There is nothing in Taweret that will break if we do it that way. The user has to be aware it might lead to overfitting of the data.

Reply:
I will take out the phrase about sources of data.

In practice, if you have only one source of data, then you can elect to do a few things:
1. Split the data set into two. Use the first split to calibrate your models and get predictions. Then use the second for the mixing phase.
2. You can use cross validation techniques to get predictions from each model. Conditional on those predictions then you can learn the weights. (I can forward some references if needed). The only draw back with this is that parameter inference becomes unclear, however you can ascertain a nice mixed prediction.

Reply:
The first approach might not be practical for me since the calibrated models have used have already used all the experimental data that are available. I would like to learn more about the second method. Would you be able to share some references? Thank you John.

Reply:
Bayesian Hierarchical Stacking: [https://arxiv.org/abs/2101.08954](https://arxiv.org/abs/2101.08954) --- They use what's called leave-one-out cross validation.

Bayesian stacking: [https://arxiv.org/pdf/1602.05162.pdf](https://arxiv.org/pdf/1602.05162.pdf) --- Also uses leave-one-out cross validation (see page 2 for brief description). This is a heavier theoretical paper, so just focus on the first few pages.

This might be good to discuss on Monday with the group. I have only seen methods from the statistician side. Would be good to hear what physicists think and do in practice.